### ISSUE TITLE
"any to one" relational field can't be edited from the app

### ISSUE THREAD

**Issue by rijkvanzanten** [directus/directus#15546]
Any to one fields (like `item` auto-generated in m2a setups) can't be edited in the app. The save button fails to engage on changes, and the relational setup doesn't properly reflect the configuration of the field. 

---

### Discussed in https://github.com/directus/directus/discussions/15545

<div type='discussions-op-text'>

<sup>Originally posted by **adanielyan** September 12, 2022</sup>
This is a follow up discussion of #15462 

I have a M2A relationship in one of my collections. In the M2A field Relationship settings I selected Delete ***/*** item (cascade) option under On Deselect of field_name. However, when I deselect the related item the record is deleted only from the Junction Collection, but not from the related collection. So the related item itself is not deleted.

I believe this can be resolved by configuring the `item` field in a junction collection to delete the related item when the record is deleted from junction collection. However the item field in junction collection cannot be edited and saved. Is there a way to unlock the system generated fields?

https://user-images.githubusercontent.com/861777/189789950-dfa0d7f0-54bd-4861-ab99-b024aa23d1eb.mov

</div>

**Comment by freekrai** [comment]
I can dig into this one

**Comment by freekrai** [comment]
So just to update: the reason we can't save any changes to that `item` field is due to missing the related collection, which it's missing since it doesn't need it so working on allowing it since the triggers dropdown means it's treated as a relational field.

**Comment by freekrai** [comment]
<img width="313" alt="Screen Shot 2022-09-20 at 11 05 51 AM" src="https://user-images.githubusercontent.com/21484/191331996-676d1f9e-4f10-4527-ade0-f93291920ff6.png">

We treat `item` as a sort of _virtual_ field if that makes sense, for example we show `Item (posts)`, `Item (tags)` (in the case of my demo app) rather than simply `Item`, which means item would never really have a related collection which means what we show in the app is correct but also makes what we want to do tricky, but still deep in the code on what we can do.

**Comment by freekrai** [comment]
Ok, coming back to this, the title isn't actually correct. You can edit "any to one" relationship fields just fine and it works as expected.

In this case, `item` is a field in the junction table and acts as a virtual field like I posted above, when we display data we use `item (tags)` / `item (posts)` / `item (any other table connected)` and since it doesn't truly have a relationship with the parent data, we can't edit it or affect it. It also doesn't know what to do with the parent table, showing the option to delete the record itself from the junction table rather than, for example, the item connected to it such as a blog post.

We treat this field as a `many two one` field in the database but don't display what it can do as it doesn't know

There are some ideas we can do:

- treat the `items` field as a system field that can't be edited and shows a lock 
- allow editing the relational triggers 
- delete the record from the junction table automatically

**Comment by freekrai** [comment]
The other option which I am leaning towards is that the `item` field in junction tables need their own UI for that special relationship which is where I will take the PR so that triggers can be edited properly.

**Comment by adanielyan** [comment]
Thank you for investigating this, Roger.

The reason I found this problem in the first place was that I was trying to delete the related "any" items when the "many" item was deleted. In my use case the "any" item can be related only to one "many" item and that many item is not used anywhere else. So when the "many" item is deleted the "any" item becomes orphan if not deleted together with "many". Deleting their relationship record from the junction table definitely makes sense in my case, but it should also cascade further and delete the "any" item. 

I just upgraded directus to 9.18.1 and everything seems to work now. Not sure if this issue is relevant anymore. I think locking the items in junction collection makes sense if everything can be controlled from the M2A field's Relationship tab.

**Comment by freekrai** [comment]
Yeah this is probably more an underlying item to look into and the locking the field might be the more simpler approach as the other method is to create a field for each relationship and let it be handled that way but wanted to give the options on how we could handle it :)

**Comment by rijkvanzanten** [comment]
Linear: [ENG-191](https://linear.app/directus/issue/ENG-191/auto-generated-any-to-one-relational-field-cant-be-edited-from-the-app)
